### PR TITLE
Restructure for monorepo (Closes #75)

### LINKED ISSUE #75: Restructure for monorepo
The tests should be shared across languages but everything else should be moved into a ruby specific directory in preparation for the first port.

### PR DESCRIPTION
**Pull request by mwhipple**

### CODE REVIEW

**Comment by mwhipple** [comment]
Travis config needs to be updated. This could be handled in a different PR since it may require a little poking around to ideally have Travis run make but still have ruby on the agent (which could then be replicated for each target runtime).

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,9 @@
+include buildsrc/gmsl
```
Comment (grsin) [inline Makefile]: What are you using from `gmsl`? I might have missed it here.

**Comment by grsin** [comment]
👍

**Review by grsin — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mwhipple) [reply Makefile]: `map` on L55

**Pull request merged**